### ISSUE TITLE
Psfp format seems support? But How to make psfp files from PBD/PHD/MID files?

### ISSUE THREAD

**Issue by Corak667** [jpd002/Play-#536]
Hi, i see you added .Psfp format seems support (even in 2015) ? But How to make psfp files from PBD/PHD/MID files? Or it's for another methods and format?
For example i ripped midi+banks from old (2004) PSP game
http://coraksoft1.narod.ru/mod/PSFP/Untold-Legends-Brotherhood-Of-The-Blade_mid-banks.7z
But how to compile them now to PSFP format that can be played on psf player?

**Comment by jpd002** [comment]
I've made some tools back then that I used to create the PSFP set for Phantasy Star Portable. They're available here: http://svn.purei.org/psfp/trunk/.

Unfortunately, I think the driver that's on there won't work with PBD/PHD/MID files, but I might have something that works at home. I thought games using "synthesized" (MIDs) music on PSP were quite rare, but if there's many games that use them, I might consider creating a tool that will work with all of them.

**Comment by Corak667** [comment]
Oh, that would be great. thank you. I don't think there's too many and it so quite popular, but some more really exist, seems people just dont digged alot. Most popular of course was at3 streaming format.
As i googled, games on people's digging that format for PSP, that used midi+pbanks results are:
Bleach Soul Carnival 2, [Untold Legends: Brotherhood Of The Blade](https://cloud.mail.ru/public/HriQ/NRK35NHMX), Persona 2 Innocent Sin, Black Rock Shooter, Sol Tigger, Shin Megami Tensei Devil Summoner

**Comment by Corak667** [comment]
Ah... It's really identical to Phantasy Star Portable .adx format you digged, but there is no "BIND" title, only "PPHD8" and midi files. So situation is almost the same, but midis are separate to single or just two banks used.

**Comment by Corak667** [comment]
Oh, you are using VS2015 code, sadly i cannot compile it, i'm using only vs2010 and cannot install vs2015..

**Comment by Corak667** [comment]
Oh, cannot compile it even with appveyor, it's "boost" dependant.
http://picsee.net/upload/2017-03-05/e96b848c6ae6.png

**Comment by jpd002** [comment]
Yeah, sorry, it's a old and non-updated project of mine. I can post a compiled executable once I get something working.

**Comment by Vaedictus** [comment]
Any updates on the status of the PSFP packer? has any executable of it been released?

**Comment by jpd002** [comment]
I've updated the packer in http://svn.purei.org/psfp/trunk/ to be easier to compile using CMake.

I've experimented with making a PSF set for Shin Megami Tensei: Devil Summoner. It takes quite a lot of work and using the packer alone isn't enough. I dunno if we can make a generic driver, I'd have to check another game (probably Untold Legends) to see what kind of code is common.

If you're interested by the PSF set, here it is:
https://purei.org/downloads/psfp/smtdevilsum.zip

**Comment by Corak667** [comment]
@jpd002 Hi
Thank you! 
Your PsfPlayer (v2014) seems don't work with this PSFp set:
"Unknown relocation type"
http://coraksoft1.narod.ru/scr/2021-11/168599690235_kiss_11kb.png 
Other builds work normally:
Phantasy_Star_Portable, ffxi_soa and xenosaga2

**Comment by jpd002** [comment]
Oh yeah, I had to make some changes to the PSP code to make SMT work properly. You can use the version available (installer) here: https://github.com/jpd002/Play-/actions/runs/5177712803

**Comment by Corak667** [comment]
yes, now works perfectly with Shin Megami Tensei - Devil Summoner

**Comment by Sukotto-1999** [comment]
Hello. Any status? If you can have time, can you make a generic driver that supports .PHD/.PBD/.MID files and make a python script or executable that packs those files to either .psfp or minipsfp/psfplib files? I posted an unplayable gamerip of Evangelion Jo on vgm.hcs64.com. I was thinking of ripping the BGM from an obscure Japanese anime game on PSP.
